### PR TITLE
Add role parameter

### PR DESCRIPTION
**Pull request by damian3031**
Added role parameter

### CODE REVIEW

File: trino/client.py [hunk 1]
```diff
@@ -238,7 +239,13 @@ def timezone(self):
...
             return self._timezone
 
-    def _format_roles(self, roles):
+    def _format_roles(self, role, roles):
+        if role and roles:
+            raise ValueError("specify either 'role' or 'roles' parameter, but not both")
+        elif role:
+            roles = {"system": role}
+        elif role is None and roles is None:
+            return {}
```
Comment (mdesmet) [inline trino/client.py:248]: I think this is no-op and can be removed.

**Review by hashhar — CHANGES_REQUESTED** [review]
Why? Why not use the `roles` mechanism instead?

**Comment by mdesmet** [comment]
> Why? Why not use the `roles` mechanism instead?

Because catalog roles are rather a legacy concept. Only hive standard SQL leverages the catalog roles. All modern access systems use a single SYSTEM role. Most users are not aware about the catalog roles concept.

Adding a single role will not confuse users but rather it will make the client more easy to use.

**Comment by damian3031** [comment]
@hashhar @mdesmet @hovaesco I refactored PR to align with discussed approach - we could pass string with only role to `roles` parameter and it is transformed to system catalog role dict.

**Review by hashhar — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -359,6 +359,18 @@ conn = trino.dbapi.connect(
...
 
+You could also pass `system` role without explicitly specifing "system" catalog:
+
+```python
+import trino
+conn = trino.dbapi.connect(
+    host='localhost',
+    port=443,
+    user='the-user',
+    roles="roleC" # equivalent to {"system": "roleC"}
```
Comment (mdesmet) [inline README.md:370]: I would advise against a mixed case example unless we are full sure it works in all cases.

Comment (hashhar) [reply README.md:370]: good point.

**Review by hovaesco — APPROVED** [review]

**Review by hashhar — APPROVED** [review]

**Pull request merged**